scout/adapter/mongo/query.py

```diff
@@ -866,7 +870,12 @@ def _get_outlier_query(query: dict) -> dict:
                     ]
                 }
 
-            outlier_padjust_query["$or"].append(pval_struct)
+            outlier_padjust_query["$or"].append(
+                {"$and": [pval_struct, {mutual_exclusion[pval_name]: {"$exists": False}}]}
```

Comment:
Enforce mutual exclusivity of the values: whenever a key/value is provided -> the other should't be existing, but if a key/value is not provided then the field might be missing - see line 878. Something like this @dnil . This is complicated 🤯

Reply:
Yes, I recall we felt a bit lucky with the clinical filter, where missing values could be handled implicitly.

Reply:
Better, but still not quite! Bugs with using just one or the other p value - on account of the `$and` they actually exclude, also when they should not (on single value query):
<img width="1038" height="517" alt="Image" src="https://github.com/user-attachments/assets/5cd5d57b-5fe0-4fe3-ae50-699cf2941bd9" />